core/shared/src/main/scala/fs2/async/mutable/Broadcast.scala

```diff
@@ -0,0 +1,218 @@
+package fs2.async.mutable
```

Comment:
I'd prefer this type (and Distribute) in either `fs2` package or `fs2.async` package. The `immutable` / `mutable` split confuses folks in my experience.

For 1.0, we could get rid of those packages even. `immutable.Signal` would become `async.ReadableSignal` or something and `mutable.Signal` would become `WritableSignal` or something, then perhaps `Signal` mixes in both interfaces?

Reply:
I agree with getting rid of them. Even if we want to keep the sub typing, it could be `ReadOnlySignal` or something

Reply:
Also, I would like to consider either remove topic, or simplify behaviour  to be similar with Broadcast

Reply:
Essentially the idea would be to define topic with broadcast and queue...

Reply:
I knew you were going to get there :P

I guess I'll have to study what `Broadcast` does more closely, but feel `Broadcast` doesn't subsume `Topic` just as `Stream` doesn't subsume `Queue`. You need to be able to "push" elements to a Stream --> Queue, you need to be able to push elements to multiple streams --> Topic, and not in every scenario the pushing comes from a stream (sometimes it's just an `F`)

Reply:
> Essentially the idea would be to define topic with broadcast and queue...

Ah, so not removing. In which case yeah, definitely up for consideration I think :)

Reply:
@SystemFw the problem with topic is that in binds too much stuff together, giving you really low control over whats going on in there. For example, it is unclear how the publish with `F` should behave when there are no consumers, or the consumers are waiting to complete the last batch.

I hope we would be able somehow fix that, w/o introducing awkward API. In worst case we can leave it as it is, but that would have two distinct implementations of similar behaviour. 

Don't get me wrong in our production code we rely on topic, but I believe we can do much better than current implementation.

Reply:
> the problem with topic is that in binds too much stuff together, giving you really low control over whats going on in there

I agree

> ut that would have two distinct implementations of similar behaviour.

I agree

> Don't get me wrong in our production code we rely on topic, but I believe we can do much better than current implementation.

I agree :)

Thanks for clarifying, we're on the same page